lift_adapter_template/lift_adapter_template/lift_adapter_template.py

```diff
@@ -119,11 +121,13 @@ def _retrieve_fail_error(value_name: str):
         new_state.current_mode = LiftState.MODE_AGV
 
         if self.lift_request is not None:
-            if self.lift_request.request_type == \
-                    LiftRequest.REQUEST_END_SESSION:
-                new_state.session_id = ''
-            else:
-                new_state.session_id = self.lift_request.session_id
+            with self.mutex:
```

Comment:
Just curious, were we facing race conditions and the fix was  to add a mutex lock for `self.lift_request`?

With python's GIL and the template running on a single process by default, we should never need to use a mutex lock

Reply:
I see, gotcha, it was added as an extreme measure but since it is unnecessary I'll remove it.

Reply:
[81a1d83](https://github.com/open-rmf/lift_adapter_template/pull/4/commits/81a1d83c276d19cf85c3df382f4391c298dd35b3)